tests/test_frontdoor.py

```diff
@@ -2112,6 +2116,46 @@ def test_epc_page_shows_epc_info():
     assert page.has_one("p:contains('23 July 2010')")
 
 
+@unittest.mock.patch("help_to_heat.frontdoor.interface.EPCApi", MockEPCApi)
+def test_epc_select_only_shows_most_recent_epc_per_uprn():
+    client = utils.get_client()
+    page = client.get("/start")
+    assert page.status_code == 302
+    page = page.follow()
+
+    assert page.status_code == 200
+    session_id = page.path.split("/")[1]
+    assert uuid.UUID(session_id)
+    _check_page = _make_check_page(session_id)
+
+    form = page.get_form()
+    form["country"] = "England"
+    page = form.submit().follow()
+
+    form = page.get_form()
+    form["supplier"] = "Utilita"
+    page = form.submit().follow()
+
+    assert page.has_text("Do you own the property?")
+    page = _check_page(page, "own-property", "own_property", "Yes, I own my property and live in it")
+
+    assert page.has_text("Do you live in a park home")
+    page = _check_page(page, "park-home", "park_home", "No")
+
+    form = page.get_form()
+    form["building_name_or_number"] = "22"
+    form["postcode"] = "FL23 4JA"
+    page = form.submit().follow()
+
+    data = interface.api.session.get_answer(session_id, page_name=address_page)
+
+    assert page.has_one("label:contains('22 Acacia Avenue, Upper Wellgood, Fulchester, FL23 4JA')")
```

Comment:
we would expect two addresses to be present? feels like this is what the below line indicates

Reply:
We expect only one instance of this visible address value. 

The line below is checking the data to ensure that neither is the excluded one on an LMK level, since we can't tell which of the above address line is shown at the current screen.

Was trying to keep the test focused on this address, given the focus of this test is about pinpointing that the newest EPC is found specifically.

Reply:
I don't think I follow this - why do we expect the other address?

Reply:
The other one should also be expected, because it's the newest of its UPRN. So really both should definitely appear if the grouping is working as expected

I'm getting a bit lost reading this whole thing back, so to reset, as the test is written now:
- The sample data contains 3 EPCs. Two for one address, one for another.
- We expect this page to show two radio button entries, one for each of the two UPRNs.
- We expect neither of the two radios has the LMK matching the old EPC for the property that has two EPCs.

This should show that we're getting the newest EPC for each UPRN.

Reply:
Ok - would it be clearer what we're testing if there was only the two for the one address in the test data? Feels like we've introduced some noise to the test without needing to: ideally unit tests should test just one thing, so there's not a risk of them failing when something else has gone wrong (making it harder to debug that thing).

Reply:
I've noticed 11 Acacia never appears, at least by address. I'm running the tests quickly to see if taking it out breaks anything.